replacy/default_match_hooks.py

```diff
@@ -140,32 +142,57 @@ def _word_is_part_of_compound_hook(doc, start, end):
     return _word_is_part_of_compound_hook
 
 
-def relative_x_is_y(arglist) -> SpacyMatchPredicate:
+def relative_x_is_y(
+    children_or_ancestors: str,
+    pos_or_dep: str,
+    value: str,
+    text: Union[str, List] = None,
+    lemma: Union[str, List] = None,
+) -> SpacyMatchPredicate:
     """
-    Written because I wanted to not match the verb `require` if its subject was clausal
+    This is a buggy, half-implementation of a DependencyMatcher, eventually
+    replaCy should allow access to spaCy's DependencyMatcher
+
+    Though it's grown, this was originally written because I wanted to
+    not match the verb `require` if its subject was clausal
     aka if it had a child with child.dep_ == 'csubj'
     So I will match_if_hook_is: false
     e.g. usage looks like
         {
             "name": "relative_x_is_y",
-            "args": [
-                "children",
-                "dep",
-                "csubj"
-            ],
+            "kwargs": {
+                "children_or_ancestors": "children",
+                "pos_or_dep": "dep",
+                "value": "csubj"
+            },
             "match_if_predicate_is": false
         }
     And this can be read as "if any children of the matched pattern have
     child.dep_ == csubj, then don't match"
     """
-    # custom patterns have to take one value as an arg, so destructure
-    children_or_ancestors, pos_or_dep, value = arglist
 
     # asserting that values are acceptable lets us template them in
     assert children_or_ancestors == "children" or children_or_ancestors == "ancestors"
     assert pos_or_dep == "pos" or pos_or_dep == "dep"
+    assert text is None or lemma is None
 
     def _relatives_x_is_y(doc, start, end):
+        def _helper(rel, pos_or_dep, value, text, lemma):
+            v = getattr(rel, f"{pos_or_dep}_", False)
+            if v == value:
```

Comment:
The code grows and I am getting confused lol
What does this function return if not `v == value` ?

Reply:
Seeing in line 212 `any()` so `True, False, None` are all ok.
Line 209 should be fine too.
But still would be awesome to return bool here. :grimacing: :pray:

Reply:
when I pulled the logic out into the `_helper` function, I dropped that case, sorry. adding back

Reply:
should be fixed now